Datadog/Example/Debugging/BackgroundEvents/BackgroundLocationMonitor.swift

```diff
@@ -0,0 +1,129 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+import Datadog
+import CoreLocation
+
+internal var backgroundLocationMonitor: BackgroundLocationMonitor?
+
+/// Location monitor used in "Example" app for debugging and testing iOS SDK features in background.
+internal class BackgroundLocationMonitor: NSObject, CLLocationManagerDelegate {
+    private struct Constants {
+        static let locationMonitoringUserDefaultsKey = "is-location-monitoring-started"
+    }
+
+    private let locationManager = CLLocationManager()
+
+    /// Tells if location monitoring is started.
+    /// This setting is preserved between application launches. Defaults to `false`.
+    ///
+    /// Note: `BackgroundLocationMonitor` can be started independently from receiving location monitoring authorization status.
+    /// Even if this value is `true`, location updates might not be delivered due to restricted or denied status.
+    var isStarted: Bool {
+        get { UserDefaults.standard.bool(forKey: Constants.locationMonitoringUserDefaultsKey) }
+        set { UserDefaults.standard.set(newValue, forKey: Constants.locationMonitoringUserDefaultsKey) }
+    }
```

Comment:
Looks like an instance var but it's actually a `static` one :) could we make this more explicit?

Reply:
I think there was one detail missing. I changed this property to `private(set)`:
```swift
private(set) var isStarted: Bool {
    get { UserDefaults.standard.bool(forKey: Constants.locationMonitoringUserDefaultsKey) }
    set { UserDefaults.standard.set(newValue, forKey: Constants.locationMonitoringUserDefaultsKey) }
}
```

Now, when looking solely on this property's code then yes, it can be static, but using instance semantic in this case gives us third option for its value: `backgroundLocationManager?.isStarted` can be `true`, `false` or `nil`.

Note that this value is **read** in `init()` to start the underlying `CLLocationManager` and it is **written** from `startMonitoring()` and `stopMonitoring()`. Instance semantic and `private(set)` ensure we have maximum cohesion in BLM state:
* if `blm?.isStarted` is `true`, we know that `CLLocationManager` is started,
* if `blm?.isStarted` is `false`, we know that `CLLocationManager` is not started,
* if `blm?.isStarted` evaluates to `nil`, we don't know the state of location monitoring.

If this is changed to `static`, then `BLM.isStarted == true` would not necessarily mean that `CLLocationManager` is running, as this thing is managed by `blm` instance.

Reply:
Yes, it makes more sense with `private(set)` 👍